aws/src/main/scala/com/gu/contentapi/client/IAMEncoder.scala

```diff
@@ -0,0 +1,41 @@
+package com.gu.contentapi.client
+
+import java.net.URLEncoder
+
+/**
+  * Requests to an IAM-authorised api-gateway must conform to AWS sig4v specification
+  */
+object IAMEncoder {
+
+  private def encodeValue(v: String) = URLEncoder.encode(v, "UTF-8").replace("+", "%20")
+
+  /**
+    * URL query params must be encoded correctly for the api-gateway authorisation to work.
+    * Note - the URL passed to `IAMSigner.addIAMHeaders` must match the URL of the request.
+    *
+    * @param params Map of query params
+    * @return Query params string
+    */
+  def encodeParams(params: Map[String, Seq[String]]): String = {
+    params.map { case (key, value) =>
+      val encoded = encodeValue(value.mkString(",")).replace("+", "%20")
+      s"$key=$encoded"
+    }.mkString("&")
+  }
+
+  /**
+    * URL query params must be encoded correctly for the api-gateway authorisation to work.
+    * Note - the URL passed to `IAMSigner.addIAMHeaders` must match the URL of the request.
+    *
+    * @param params Query params string
+    * @return Query params string
+    */
+  def encodeParams(params: String): String = {
+    params.split("&").toList.flatMap { s =>
+      s.split("=").toList match {
```

Comment:
`.toList.collect` can save you a few keystrokes 😓

Reply:
hmm, `s.split("=").toList.collect { ...` will call the pf for each string in the list. I'm using the `match` to check that the list is a pair of strings, not sure if there's a better way

Reply:
I think this will work
```scala
s.split("=").toList.collect { case k :: v :: Nil => s"..." }
```

Reply:
this fails to compile because `collect` iterates through the collection, so:
`s.split("=").toList.collect { case s: String => ... }`

Reply:
yes of course 😅

Reply:
I knew there was something, you just need to bubble the call one level up:
```scala
params.split("&")
  .map(_.split("="))
  .collect { case Array(k, v) => s"${encode(k)}=${encode(v)} }
  .mkString("&")
```

Reply:
Nice, that's a bit clearer!